### PR TITLE
Fix expand margin on TextEdit StyleBoxes, and add scrollbar padding

### LINKED ISSUE #74445: Expand Margin is not working in Styles for TextEdit
### Godot version

4.1.dev (5dccc94)

### System information

Ubuntu 22.04, Vulkan renderer, gforce 1650ti

### Issue description

Modifying the Expand Margin property in a style applied to a TextEdit does not draw extending outside the rect like in other controls, which causes the style to be clipped.

### Steps to reproduce

1. Add a TextEdit
2. Add a Style to it
3. Use expand margins
4. Note that it doesn't work as expected

### Minimal reproduction project

[TestTheme.zip](https://github.com/godotengine/godot/files/10892436/TestTheme.zip)

### PR DESCRIPTION
**Pull request by azagaya**
This fixes #74445, and also adds padding constants for the scroll bars in the theme for TextEdits, as the scrollbars always collide with the border.

This is the way it would work with this PR

https://user-images.githubusercontent.com/46932830/222998953-5d015475-727d-4918-b0f7-9d545a21001f.mp4

The padding constants were implemented in the theme, because most of the times them clashing with the theme is the reason why one would like to move them.

### CODE REVIEW

**Comment by Chaosus** [comment]
Need to be rebased in order to fix CI fails.

File: scene/gui/text_edit.cpp [hunk 1]
```diff
@@ -511,7 +511,7 @@ void TextEdit::_notification(int p_what) {
 			_update_scrollbars();
 
 			RID ci = get_canvas_item();
-			RenderingServer::get_singleton()->canvas_item_set_clip(get_canvas_item(), true);
+			RenderingServer::get_singleton()->canvas_item_set_clip(get_canvas_item(), false);
```
Comment (Calinou) [inline scene/gui/text_edit.cpp:514]: See https://github.com/godotengine/godot/issues/32996 – this may cause other issues.

**Comment by akien-mga** [comment]
@azagaya Did you mean to close this? It seems like you pushed 0 commits (no change) to your `master` branch, from which you made this PR, so it automatically closed the PR (and it can't be reopened after this).

For reference, your previous commit is there: https://github.com/godotengine/godot/commit/aa5dfa60b70ad94f4e9d554edda6772a67ce0b52

**Comment by azagaya** [comment]
Hey akien, no it wasnt intentional, i just moved my changes on my fork from master to other branch, because i wanted to sync master to test other PR.

Anyways i dont think ill be able to fix this anytime soon. I managed to fix the expand border issue, but that caused wrong clipping of the text. I fear the issue is much deeper than i intially thought. If anyone has any hints of course i can look at this again. But for now this could remain closed imho.

**Pull request closed without merging**